clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -717,13 +1023,29 @@ private Struct listOffsetsResponse(Map<TopicPartition, Long> offsets, short erro
         return new ListOffsetResponse(partitionData).toStruct();
     }
 
-    private Struct fetchResponse(TopicPartition tp, long fetchOffset, int count) {
+    private Struct fetchResponse(Map<TopicPartition, FetchInfo> fetches) {
+        Map<TopicPartition, PartitionData> tpResponses = new HashMap<>();
+        for (Map.Entry<TopicPartition, FetchInfo> fetchEntry : fetches.entrySet()) {
+            TopicPartition partition = fetchEntry.getKey();
+            long fetchOffset = fetchEntry.getValue().offset;
+            int fetchCount = fetchEntry.getValue().count;
+            MemoryRecords records = MemoryRecords.emptyRecords(ByteBuffer.allocate(1024), CompressionType.NONE);
+            for (int i = 0; i < fetchCount; i++)
+                records.append(fetchOffset + i, 0L, ("key-" + i).getBytes(), ("value-" + i).getBytes());
+            records.close();
+            tpResponses.put(partition, new FetchResponse.PartitionData(Errors.NONE.code(), 0, records.buffer()));
+        }
+        FetchResponse response = new FetchResponse(tpResponses, 0);
+        return response.toStruct();
+    }
+
+    private Struct fetchResponse(TopicPartition partition, long fetchOffset, int count) {
```

Comment:
Can this be implemented just by delegating to the generic version with a singleton map?